### PR TITLE
[release process] Add a github actions to backport fix automatically

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :scroll: Docs          |

## Description

Based on https://github.com/tibdex/backport which seems awfully conveniant.

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Backport
+on:
+  pull_request_target:
+    types:
+      - closed
+      - labeled
```
Comment (DanielNoord) [inline .github/workflows/backport.yml:6]: I guess this should only be closed then? If we only want to run on merged PRs?

Comment (Pierre-Sassoulas) [reply .github/workflows/backport.yml:6]: It's taken from step 1 in https://github.com/tibdex/backport#usage directly

Comment (DanielNoord) [reply .github/workflows/backport.yml:6]: I mean, I haven't tested this, but it feels a bit weird. It should only run whenever a pull request gets closed, see:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Comment (Pierre-Sassoulas) [reply .github/workflows/backport.yml:6]: Now that I think about it, it's probably so you can backport after merging by adding labels. (Which would have avoided the work in #7775 )

Comment (DanielNoord) [reply .github/workflows/backport.yml:6]: Makes sense!

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name: Backport
+    runs-on: ubuntu-latest
+    # Only react to merged PRs for security reasons.
+    # See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target.
+    if: >
+      github.event.pull_request.merged && (
+        github.event.action == 'closed'
+        || (
+          github.event.action == 'labeled'
+          && contains(github.event.label.name, 'backport')
```
Comment (DanielNoord) [inline .github/workflows/backport.yml:19]:
```suggestion
          && contains(github.event.label.name, 'needs backport')
```

Comment (Pierre-Sassoulas) [reply .github/workflows/backport.yml:19]: The github action's code force us to use this keyword. Also as it's done automatically we don't "need" to do anything 🎉

**Review by DanielNoord — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Seems like I still need to rune up the config because of ``tibdex/backport@v2 is not allowed to be used in PyCQA/pylint. Actions in this workflow must be: within a repository owned by PyCQA, created by GitHub, or matching the following: kanga333/comment-hider@*``. I reviewed the code of the action itself and I did not see anything nefarious looking. But it's not an official github action.

**Pull request merged**